transient-aws/hive-example/dispatch.sh

```diff
@@ -0,0 +1,316 @@
+
+terminate() {
+    for i in {5..0}; do
+        echo -ne "Terminating cluster ${CLUSTER_NAME} in $i seconds"'\r'
+        sleep 1
+    done
+
+    #${WGET} --method DELETE "${CLUSTER_URL}"
+    # Standard Centos version of wget doesn't support --method flag
+
+    curl -u ${DIRECTOR_USER}:${DIRECTOR_PASSWORD} -X "DELETE" "${CLUSTER_URL}"
+
+
+    STAGE=$(${WGET} -q "${CLUSTER_URL}/status" | jq -r ".stage");
+    if [[ "${STAGE}" != "TERMINATING" ]]; then
+        echo "Unable to terminate cluster. Stage is ${STAGE}."
+        echo "Please terminate cluster ${CLUSTER_NAME} manually!"
+        exit -1
+    fi
+
+    while
+        STAGE=$(${WGET} -q "${CLUSTER_URL}/status" | jq -r ".stage");
+        [ "${STAGE}" == "TERMINATING " ];
+    do
+        echo "Cluster ${CLUSTER_NAME} is terminating. Waiting for 10 seconds ..."
+        sleep 10
+    done
+
+    echo "Cluster ${CLUSTER_NAME} is terminated"
+
+    if [[ "${KEEP_CM}" == "false" ]]; then
+        ENV_URL="${DIRECTOR_SERVER}/api/v4/environments/${ENVIRONMENT_NAME}/deployments/${DEPLOYMENT_NAME}/"
+        echo "Terminating CM Instance"
+
+         #${WGET} --method DELETE "${ENV_URL}"
+        curl -u ${DIRECTOR_USER}:${DIRECTOR_PASSWORD} -X "DELETE" "${ENV_URL}"
+
+         while
+            STAGE=$(${WGET} -q "${ENV_URL}/status" | jq -r ".stage");
+            [ "${STAGE}" == "TERMINATING" ];
+        do
+            echo "Environment ${DEPLOYMENT_NAME} is terminating. Waiting 10 seconds"
+            sleep 10
+        done
+    fi
+    echo "Deployment ${DEPLOYMENT_NAME} is terminated"
+
+}
+
+#
+# Import and job dispatch logic
+#
+
+# Provision the cluster if not in job-submission mode
+PROVISIONED="false"
+if [[ -f "${PROVISION_CONFIG}" ]]; then
+    # Path to config file with all the details needed to create the cluster
+
+    # Import client config to Director
+    PROVISIONED="true"
+    echo "Importing client config ${PROVISION_CONFIG} to ${DIRECTOR_SERVER}"
+
+    ${WGET} --header="Content-Type: text/plain" --post-file "${PROVISION_CONFIG}" \
+        "${DIRECTOR_SERVER}/api/v4/import?clusterName=${CLUSTER_NAME}&deploymentName=${DEPLOYMENT_NAME}&environmentName=${ENVIRONMENT_NAME}"
+    echo
+
+    # Iterate on status endpoint for cluster based on response
+
+    echo "Waiting for cluster ${CLUSTER_NAME} to get to READY stage ..."
+
+    progress="0"
+    while
+        STAGE=$(${WGET} -q "${CLUSTER_URL}/status" | jq -r ".stage");
+        [ "${STAGE}" != "READY" ];
+    do
+        if [ "${STAGE}" == "BOOTSTRAP_FAILED" ]; then
+            echo "Cluster ${CLUSTER_NAME} failed to  bootstrap. Please check server logs."
+            exit -1
+        fi
+        if [[ -z "${STAGE}" ]]; then
+            echo "Unable to retrieve status for cluster ${CLUSTER_NAME}. Please check server logs."
+            exit -1
+        fi
+        echo "Stage is ${STAGE}. Progress: ${progress}%..."
+        if [ "$progress" -lt "100" ]; then
+           let "progress+=1"
```

Comment:
Hahah! I got caught red handed.  This was meant to be removed before going out.  It's useful while testing.